Support Composition rendering

edwardjcruz opened the pull request:
hey @seanno I added support to render the data sources sent in the IPS from the Composition resource. 

<img width="824" alt="Screenshot 2023-12-08 at 4 33 46 PM" src="https://github.com/the-commons-project/shc-web-reader/assets/38390406/5afb0176-8653-439b-8382-e0f4a4e6c74a">

Also added an error for when the data bundle is empty as we came across this case during testing with Yuri. 

<img width="883" alt="Screenshot 2023-12-08 at 4 34 14 PM" src="https://github.com/the-commons-project/shc-web-reader/assets/38390406/343d7f64-96dc-44f5-a4f3-16675debfb45">

seanno reviewed (changes requested):
Hey Eddie, sorry so late with this! The data missing part looks great. For the composition text, two things: (1) it seems like we should make this row conditional on presence, don't you think? i.e., we may get an IPS without this content and better to just have no row vs. an empty one; (2) I'd like to stick to one way to sanitize inbound HTML ... can you use the dompurify/iframe approach for this as we do in PatientSummarySection.js:renderNarrative, rather than the html-react-parser approach? this will keep the bundle size down and I spent a bunch of time trying to make sure we were conservative in how we rendered html coming from a potentially untrusted source.

edwardjcruz commented:
Hey @seanno thank you for the feedback! I removed the use of the html-react-parser library, switched over to utilizing dompurify/iframe, also made it a conditional row!
